### ISSUE TITLE
Need help, Putty 7 and JAWS 19, unable to use vi ditor because of lack of navigation feedback.

### ISSUE THREAD

**Issue by SAMorgan258** [munawarb/JAWS-PuTTY-Scripts#1]
Hello,
I was forced to give up Exceed 14 (last version  to work on Windows 7), and am using Putty on Windows 10.
I am able to login and issue command and read output using Putty and JAWS 19.  But when in "vi" editor, none of the characters to move around like h, j, k, l for left, down, up, and right announce what character you are over.

I can edit the, but if I do a typo, there is no way to correct if I make a mistake.

What version of Putty did the JAWS scripts work with?

**Comment by munawarb** [comment]
Hi @SAMorgan258, The reason these keys are not working as expected is because these JAWS scripts redefine behavior specifically for the arrow keys. I never implemented JAWS-specific behavior for the keys you mentioned. Can you try your luck with Nano and see how it works out for you? I don't think there's been a PuTTY update for a while now; if there has been, PuTTY doesn't tend to release versions that break things like JAWS scripts and NVDA add-ons.

**Comment by SAMorgan258** [comment]
Ks for reaching out.

I am a systems integrator doing infrastructure on various UNIX/Linux platforms and not all UIX operating systems have nano.

I am not familiar with “nano” because I am constantly using “ssh” to move around to various systems.  Currently I am forced to copy files back to a Samba server to edit the files using an editor on my PC, and then copy the file back to the system.  This really slows things down.

The “vi” editor is the one editor I know will be on all UNIX/Linux systems.  But after being forced to upgrade to Windows 10, I am having to find a new way of working.

Can you recommend a different terminal emulator to SSH to other systems that might work better with “vi”?

Exceed version 15 went to graphical characters and is no longer compatible with JAWS.

Sam

**Comment by SAMorgan258** [comment]
Thanks, I found “nano” on most of the systems I support.  So I will learn how to use it.  From Google, it does not seem as flexible as “vi”, but it should get the job done.

Sam

**Comment by munawarb** [comment]
Hi @SAMorgan258, that's disappointing that version 15 is not accessible. You can probably install a client on a Windows Linux Subsystem for Windows 10 and ssh from there. Otherwise, PuTTY is the only client I know of that's actively supported on Windows.

You could also try switching to NVDA while you work with SSH and then switch back to JAWS. I've had great luck with NVDA and terminal sessions. I don't have JAWS installed on this system so can't help debug the issue.

I'm glad you found Nano to be of some use. I use this editor full-time whenever I need to SSH to a system. Hopefully it works out to be a good alternative.
